numpy/lib/format.py

```diff
@@ -468,18 +474,18 @@ def _read_array_header(fp, version):
     # header.
     import struct
     if version == (1, 0):
-        hlength_str = _read_bytes(fp, 2, "array header length")
-        header_length = struct.unpack('<H', hlength_str)[0]
-        header = _read_bytes(fp, header_length, "array header")
+        hlength_type = '<H'
```

Comment:
Dogfooding by using `dtype('<H')` might also be neat, and then `np.frombuffer(hlength_str, dtype=...)